src/Compilers/Core/Portable/InternalUtilities/RoslynLazyInitializer.cs

```diff
@@ -19,11 +19,11 @@ internal static class RoslynLazyInitializer
             => LazyInitializer.EnsureInitialized<T>(ref target!, valueFactory);
 
         /// <inheritdoc cref="LazyInitializer.EnsureInitialized{T}(ref T, ref bool, ref object)"/>
-        public static T EnsureInitialized<T>([NotNull] ref T? target, ref bool initialized, [NotNull] ref object? syncLock)
+        public static T EnsureInitialized<T>([NotNull] ref T? target, ref bool initialized, [NotNullIfNotNull("syncLock")] ref object? syncLock)
```

Comment:
That's clever ;-)
Do we still need `RoslynLazyInitializer` now that `LazyInitializer` is annotated? If it's not a trivial change, please file an issue and assign it to me.

Reply:
I copied the annotation from net5.0.
Lack of `[NotNull]` should have impact on netstandard2.0

Reply:
Makes sense. Thanks